### PR TITLE
Add Teleport TKL

### PR DESCRIPTION
**Pull request by ebastler**
This PR adds support for the Teleport TKL PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/teleport/tkl/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_tkl_all(
+        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,             KC_PSCR, KC_SCRL, KC_PAUS,
+
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,   KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                       KC_UP,
```
Comment (zvecr) [inline keyboards/teleport/tkl/keymaps/via/keymap.c:14]: You have `MO(1)` but only a single layer. Either this needs to be removed, or an additional layer added.

File: keyboards/teleport/tkl/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_tkl_all(
+        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,             KC_PSCR, KC_SCRL, KC_PAUS,
+
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,   KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                       KC_UP,
```
Comment (zvecr) [inline keyboards/teleport/tkl/keymaps/default/keymap.c:14]: Same here

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/teleport/tkl/keymaps/via/keymap.c (hunk 1 above)
Comment (ebastler) [reply keyboards/teleport/tkl/keymaps/via/keymap.c:14]: Oops, I reused the base keymaps of another TKL of mine. Will remove.

Comment (ebastler) [reply keyboards/teleport/tkl/keymaps/via/keymap.c:14]: I decided to instead add a NKRO toggle to layer 1 - should be fine now, I hope. Also some minor fixes to make layouts appear correctly in configurator.

File: keyboards/teleport/tkl/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+* Hardware Supported: The Teleport TKL PCB, Rev 1
+* Hardware Availability: [The Teleport](https://www.theteleport.de/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make the_teleport_tkl:default
+
+Flashing example for this keyboard:
+
+    make the_teleport_tkl:default:flash
```
[the remark this answers is not in the document]
Comment (ebastler) [reply keyboards/teleport/tkl/readme.md]: Thank you for catching this one - I generated a template first, then added it to existing teleport folder and renamed accordingly. This escaped my renaming...

**Review by waffle87 — APPROVED** [review]

**Pull request merged**